### PR TITLE
Add API to benchmark allowing for custom context to be added

### LINKED ISSUE #525: Custom context fields
I am interested in adding custom context fields to benchmarks. As far as I can tell this isn't possible right now? If not would it be possible to add?

The use case I have in mind is adding our software build version to the context. This way output JSON/CSV data files can be passed around or stored and contain the information required to reproduce results.

Not critical since this information can be stored elsewhere. But it might be nice. Thoughts?

### PR DESCRIPTION
**Pull request by dominichamon**
Fixes #525

### CODE REVIEW

File: src/reporter.cc [hunk 1]
```diff
@@ -71,6 +76,12 @@ void BenchmarkReporter::PrintBasicContext(std::ostream *out,
     Out << kv.first << ": " << kv.second << "\n";
   }
 
+  if (internal::global_context != nullptr) {
```
Comment (LebedevRI) [inline src/reporter.cc:79]: I was actually expecting that `FLAGS_benchmark_context` would be parsed into `internal::global_context`.

Comment (dominichamon) [reply src/reporter.cc:79]: oh that would be smart.

**Review by LebedevRI — COMMENTED** [review]
Looks about right.
Tests missing.
Also, i think you may want to explicitly spell out in the docs, and test,
what happens when there's key collision.

**Comment by dominichamon** [comment]
> Looks about right.
> Tests missing.
> Also, i think you may want to explicitly spell out in the docs, and test,
> what happens when there's key collision.

done, done, done :)

**Review by LebedevRI — APPROVED** [review]

**Comment by dominichamon** [comment]
waiting for CI. Thanks, @LebedevRI.

**Pull request merged**